src/ucs/datastruct/list.h

```diff
@@ -121,6 +123,8 @@ static inline int ucs_list_is_empty(const ucs_list_link_t *head)
 static inline int
 ucs_list_is_first(const ucs_list_link_t *head, const ucs_list_link_t *elem)
 {
+    UCS_LIST_CHECK_ELEM_IS_VALID(head);
+    UCS_LIST_CHECK_ELEM_IS_VALID(elem);
```

Comment:
no need to check elem, we deref it
we need to check elem->prev
same comment for ucs_list_is_last